procurement_groupby_analytic/models/procurement.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Authors: Laetitia Gangloff
+#    Copyright (c) 2015 Acsone SA/NV (http://www.acsone.eu)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+
+from openerp import api, fields, models
+
+
+class ProcurementOrder(models.Model):
+    _inherit = 'procurement.order'
+
+    account_analytic_id = fields.Many2one(
+        related='sale_line_id.order_id.project_id',
+        comodel_name='account.analytic.account')
+
+    @api.model
+    def _get_available_draft_po_domain(self, procurement, partner):
+        """ If account analytic is defined on procurement order
+            search draft purchase order on this criteria
+        """
+        available_draft_po_domain = super(
+            ProcurementOrder, self)._get_available_draft_po_domain(
+                procurement, partner)
+        available_draft_po_domain.append(
```

Comment:
I don't think this should be the criteria to link sales orders with an analytic account with purchase orders. Follow the procurement chain for this.